patches/tModLoader/Terraria/ModLoader/ModMenu.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria.GameContent;
+using Terraria.UI;
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// A class that is used to customize aesthetic features of the main menu, such as the logo, background and music.
+	/// </summary>
+	public abstract class ModMenu : ModType
+	{
+		internal UserInterface userInterface;
+
+		/// <summary>
+		/// The amount of time (in ticks) that this ModMenu has been active for. Resets when the menu is not selected.
+		/// </summary>
+		public int timeActive;
+
+		protected override void Register() {
+		}
```

Comment:
Why is this not used? Please move all your hardcode from ModLoader.cs to a loader class that functions through this override.